File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        d={}
        for i in range(len(nums)):
            if target-nums[i] in list(d.keys()):
                return [i,d[(target-nums[i])]]
            else:
                d[nums[i]]=i

File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> numMap;
        for(int i=0; i<nums.size();i++){
            numMap[nums[i]]=i;
        }
        for(int i=0;i<nums.size();i++){
            int remaining=target-nums[i];
            if(numMap.count(remaining) && numMap[remaining]!=i){
                return {i,numMap[remaining]};
            }
        }
        return {};
    }
};